content/docs/rfcs/48/README.md

```diff
@@ -0,0 +1,104 @@
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+## Goals
+
+This specification is intended to formally document the names and expected behaviour of the RADIO-DISH socket types, which together form the ZeroMQ radio-dish pattern. Conforming implementations of these sockets SHOULD respect this specification, thus ensuring that applications can depend on predictable behavior. This specification is not transport specific, but not all behaviour will be reproducible on all transports.
+
+## Overall Goals of this Pattern
+
+Radio-dish pattern is member of a new family of thread-safe sockets.
+The raido-dish pattern is the thread-safe alternative of the pubsub pattern.
+
+The pattern is intended for event and data distribution, usually from a small number of publishers (radios) to a large number of subscribers (dishes), but also from many publishers to a few subscribers.
+
+Group membership checks can either happen on the RADIO or DISH side and depend on the transport protocol and are defined in the relevant documents. For TCP, refer to http://rfc.zeromq.org/spec:37/ZMTP.
+
+In order for the API to be thread-safe sending and receiving messages MUST be atomic and a single API call to receive or send entire msg. Therefore, the radio-dish pattern (and the rest of the thread-safe family) MUST NOT allow multipart messages.
+
+## The RADIO Socket Type
+
+The RADIO socket type provides basic one-way broadcasting to a set of dishes. Over TCP, it does filtering on outgoing messages but nonetheless a message will be sent multiple times over the network to reach multiple dishes. RADIO is used mainly for transient event distribution where stability of the network (e.g. consistently low memory usage) is more important than reliability of traffic.
+
+General behavior:
+
+* MAY be connected to any number of DISH sockets, and SHALL only send messages.
+* SHALL maintain a single outgoing message queue for each connected socket.
+* SHALL create a queue when initiating an outgoing connection to a socket, and SHALL maintain the queue whether or not the connection is established.
+* SHALL create a queue when a socket connects to it. If this socket disconnects, the RADIO socket SHALL destroy its queue and SHALL discard any messages it contains.
+* SHOULD constrain queue sizes to a runtime-configurable limit.
+* SHALL silently discard any messages that dishes send it.
+* MUST be thread-safe and allow sending from multiple threads.
+* MUST allow attaching a group to a message in a thread-safe way.
+
+For processing outgoing messages:
+
+* SHALL not modify outgoing messages in any way.
+* MAY, depending on the transport, send all messages to all dishes.
```

Comment:
should these differences based on transport be defined as well?

Reply:
I documented it in ZMTP 3.1, we may have a link here to the section in ZMTP 3.1.
We can have another one when we will have the UDP rfc.

Reply:
great, thanks